pre_commit/commands/hook_impl.py

```diff
@@ -111,49 +112,86 @@ def _pre_push_ns(
     remote_name = args[0]
     remote_url = args[1]
 
+    # TODO(zombiezen): This only triggers hooks for a single ref,
+    # instead of all refs pushed.
     for line in stdin.decode().splitlines():
-        local_branch, local_sha, remote_branch, remote_sha = line.split()
+        local_ref, local_sha, remote_ref, remote_sha = line.split()
```

Comment:
Nit: I think I'd prefer to not rename these to keep them more similar to upstream/potentially other uses in the codebase. How are `ref` and `branch` used elsewhere in pre-commit? In my mind it's more important that they be consistent than maximally make sense in this particular spot.

Reply:
So pre-commit uses the following terminology:

- "Ref" refers to what Git calls a "commit hash"
- "Branch" refers to what Git calls a "ref name", which may not necessarily be a branch

I can revert if you like, but I did this because otherwise cross-referencing the code for correctness with the [Git docs](https://git-scm.com/book/en/v2/Customizing-Git-Git-Hooks) gets pretty confusing.

Reply:
I don't feel super strongly about it, but my preference would be for these names to remain consistent with the rest of the codebase.